src/lsm/grid.zig

```diff
@@ -536,17 +542,18 @@ pub fn GridType(comptime Storage: type) type {
                 grid.read_resolving = false;
             }
 
+            // Remove the "root" read so that the address is no longer actively reading / locked.
+            grid.read_queue.remove(read);
+
             // Resolve all reads queued to the address with the block.
             // Callbacks may queue more to read.resolves so it must drain any new/existing pendings.
```

Comment:
Note that this should no longer be true since the "root" read was removed from the read_queue, but otherwise looks good.

Reply:
Good point.